pages/views.py

```diff
@@ -14,6 +14,12 @@ class ViewPage(UserPassesTestMixin, generic.DetailView):
     context_object_name = 'page'
     template_name = "pages/page.html"
 
+    def get_context_data(self, **kwargs):
```

Comment:
I wouldn't do this here (as it makes it less general). This kind of check should be done in the template